runtime/syntax/vim.vim

```diff
@@ -570,21 +570,9 @@ if !filereadable(s:luapath)
   endif
  endfor
 endif
-if (g:vimsyn_embed =~ 'l' && has("lua")) && filereadable(s:luapath)
```

Comment:
`runtime/doc/syntax.txt` has two paragraphs on these variables, `g:vimsyn_embed` and `g:vimsyn_folding`. The language specific docs there (lua, ruby, etc.) should be removed since they now don't exist any more.